specifyweb/frontend/js_src/lib/components/QueryBuilder/CreateRecordSet.tsx

```diff
@@ -109,3 +115,30 @@ export function CreateRecordSet({
     </>
   );
 }
+
+/**
+ * Create a record set from selected records.
+ * See also `MakeRecordSetButton`
+ */
+export function CreateRecordSetFromQuery({
+  getIds,
+  baseTableName,
+  queryResource,
+}: {
+  readonly getIds: () => RA<number>;
+  readonly baseTableName: keyof Tables;
+  readonly queryResource: SpecifyResource<SpQuery> | undefined;
+}): JSX.Element {
+  const recordSetName =
+    queryResource === undefined || queryResource.isNew()
+      ? undefined
+      : queryResource.get('name');
```

Comment:
(optional)

why does CreateRecordSetFromQuery exist?
is it just for these 3 lines?
the rest is just proxying the props to CreateRecordSet

can you just call CreateRecordSet directly - that would take less code that this

the only reason for CreateRecordSetFromQuery to exist is if you want to add some documentation for it, that is not applicable for CreateRecordSet

Reply:
> can you just call CreateRecordSet directly - that would take less code that this

Fixed in https://github.com/specify/specify7/pull/4548/commits/b844b88dd40bfd6544237b010650775e653bebff